src/main/java/uk/gov/hmcts/reform/blobrouter/services/NewEnvelopesFinder.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.blobrouter.services;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.data.EnvelopeRepository;
+
+import java.time.Instant;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.time.temporal.ChronoUnit.HOURS;
+import static java.util.Collections.singleton;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class NewEnvelopesFinder {
+
+    private static final Logger logger = getLogger(NewEnvelopesFinder.class);
+
+    private final EnvelopeRepository envelopeRepository;
+
+    private final ServiceConfiguration serviceConfig;
+
+    private static final String CRIME_CONTAINER = "crime";
+
+    public NewEnvelopesFinder(
+        EnvelopeRepository envelopeRepository,
+        ServiceConfiguration serviceConfig
+    ) {
+        this.envelopeRepository = envelopeRepository;
+        this.serviceConfig = serviceConfig;
+    }
+
+    public void checkNewCftEnvelopesCreated() {
+        checkNewEnvelopesInContainers(getCftContainers(), "CFT");
+    }
+
+    public void checkNewCrimeEnvelopesCreated() {
+        checkNewEnvelopesInContainers(singleton(CRIME_CONTAINER), "CRIME");
```

Comment:
here we are not checking if crime is enabled or not ?

Reply:
Good point, I'll update